libs/designer/src/lib/core/templates/utils/helper.ts

```diff
@@ -84,6 +74,56 @@ export const getFilteredTemplates = (
     });
     return hasDetailFilters;
   });
+
+  const searchOptions = {
+    isCaseSensitive: false,
+    includeScore: false,
+    threshold: 0,
+    ignoreLocation: true,
+    keys: [
+      'title',
+      'description',
+      {
+        name: 'manifest',
+        weight: 2,
+        getFn: ([_name, template]: [string, Template.Manifest]) => [
+          ...template.skus,
+          ...(template.tags ?? []),
+          ...(template.kinds ?? []),
+          ...Object.values(template.details),
+        ],
+      },
+      {
+        name: 'parameters',
+        weight: 3,
+        getFn: ([_name, template]: [string, Template.Manifest]) =>
+          template.parameters?.reduce((acc: string[], parameter) => acc.concat([parameter.displayName, parameter.description]), []),
+      },
+      {
+        name: 'connections',
+        weight: 3,
+        getFn: ([_name, template]: [string, Template.Manifest]) =>
+          template.connections
+            ? Object.values(template.connections)?.reduce(
+                (acc: string[], connection) =>
+                  acc.concat([
+                    connection.connectorId, // TODO: change after PR (connection name to be stored in the state)
```

Comment:
What do you mean by connection name getting stored in state?, we only store connection id and its display name